### ISSUE TITLE
Screws up card order after reposition

### ISSUE THREAD

**Issue by lovac42** [Arthur-Milchior/anki-correct-due#1]
Arthur, thanks for bring this to attention.

The SQL query may not be correct:
```
    cids = col.db.list(f"select id from cards order by id and type = {CARD_NEW}")
```
I changed it to (python 2.7):
```
    # Save time
    if not col.db.scalar("select id from cards where type=0 and due>666000"):
        return

    # Sorted by due and siblings for whole collection
    cids = col.db.list("select id from cards where type=0 order by due,ord")
    sched.sortCards(cids,shift=True)
```
This will keep the existing order of cards. It can also wrap around ```anki.collection._Collection.fixIntegrity``` to auto check whenever the db was checked.

Hope this helps.

**Comment by Arthur-Milchior** [comment]
The SQL query works, so it seems to be correct sql. So I mean that's not actually what you mean by «not be correct».
However, I agree with you, it may change the card order. In particular because it recomputes a random order for random deck.

I should however note that your solution may also change some card order. This is the case as soon as two cards of the same note have different due values. This may occur if both cards are in distinct decks, and one deck's option is switch to randomize or to «added order». 

Actually, my query ensure that cards in non-random decks are REALLY shown in the order in which they are created. Which is often not the case. I actually just wrote this document https://github.com/Arthur-Milchior/anki/blob/master/documentation/due.md to explain all the way I know in which due values may be wrong.

**Comment by lovac42** [comment]
Good point about different decks. I should bypass the API to reorder this. In my case, some decks have custom sorting orders that is not related to creation time.

**Comment by Arthur-Milchior** [comment]
I guess I'll add a configuration option to decide which of the two reordering you may want. It would clearly make sens. And I should merge this in the add-on (Check Database)[https://ankiweb.net/shared/info/1135180054]

**Comment by lovac42** [comment]
If a user runs your addon before checking db, these cards would be sorted by 32bit system time. Then using ```due``` would make sense as it keeps the order of existing cards. But for cards that's already truncated the value to 1000000, the due field would be equal then it should be sorted by id. My idea is to have your addon run before each database checkup, so sorting by ```due,id``` would be a better choice.

Here is how I modified it:  https://github.com/lovac42/anki-correct-due/blob/master/redue.py
